brian2/groups/group.py

```diff
@@ -54,17 +54,24 @@ def _display_value(obj):
     value : str
         A string representation of the object
     '''
+    if isinstance(obj, Function):
+        return '<Function>'
     try:
-        return repr(obj.get_value())
+        obj = obj.get_value()
     except AttributeError:
         pass
     try:
-        return repr(obj.value)
+        obj = obj.value
     except AttributeError:
         pass
-    if isinstance(obj, Function):
-        return '<Function>'
-    return repr(obj)
+
+    # We (temporarily) set numpy's print options so that array with more than
+    # 10 elements are only shown in an abbreviated way
+    old_options = np.get_printoptions()
+    np.set_printoptions(threshold=10)
+    str_repr = repr(obj)
+    np.set_printoptions(**old_options)
```

Comment:
Is there a case for using a try/except here just in case there's an error raised in the repr it would leave numpy's print options in a different state (which might be annoying if the user was catching exceptions). Slightly obscure possibility I guess, not sure if worth worrying about.

Reply:
If this were a general display function, then I'd agree that this is important. This `_display_value` function however is only used in the warning for ambiguous identifiers, and this warning is only raised for variables of types that could potentially be referred to from equations. We previously filter out everything which is not one of `numbers.Number, np.ndarray, np.number, Function, Variable`, so calling `repr` here should be safe. But then, better safe then sorry: I'll add a quick try/except/final :)